### PR TITLE
[stable/jenkins] Added jenkins-master ingress support

### PR DESCRIPTION
**Pull request by electroma**
Added option to create Ingress object for Jenkins master.
May be useful in case server uses Ingress controller.

### CODE REVIEW

**Comment by lachie83** [comment]
@electroma Thanks for the contrib. Just out of interest. What ingress controllers did you test this against?

**Comment by electroma** [comment]
@lachie83 Tested with Traefik (chart stable/traefik).

**Comment by mgoodness** [comment]
@lachie83 @electroma I'm going to give it a shot against the nginx controller. Will report back.

File: stable/jenkins/values.yaml [hunk 1]
```diff
@@ -17,7 +17,10 @@ Master:
 # JavaOpts: "-Xms512m -Xmx512m"
   ServicePort: 8080
 # For minikube, set this to NodePort, elsewhere use LoadBalancer
+# Use ClusterIP if your setup includes ingress controller
   ServiceType: LoadBalancer
+# Used to create Ingress record (should used with ServiceType: ClusterIP)
+  HostName: ~
```
Comment (viglesiasce) [inline stable/jenkins/values.yaml]: Looks like this is being set by default. Shouldnt it be commented out?

File: stable/jenkins/templates/jenkins-master-ingress.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{{- if .Values.Master.HostName }}
+---
```
Comment (viglesiasce) [inline stable/jenkins/templates/jenkins-master-ingress.yaml]: Remove this otherwise delete will fail.

**Review by viglesiasce — DISMISSED** [review]
Needs version bump as well

**Comment by electroma** [comment]
@viglesiasce fixed according to your comments

File: stable/jenkins/values.yaml (hunk 1 above)
Comment (electroma) [reply stable/jenkins/values.yaml]: I would prefer to have this key uncommeted, but empty. If there is a convention to comment out empty keys - lets stick to it.

**Comment by electroma** [comment]
@viglesiasce does it qualify as minor change (so the new version is 0.1.8)?

**Comment by viglesiasce** [comment]
@k8s-bot ok to test

File: stable/jenkins/templates/jenkins-master-ingress.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{{- if .Values.Master.HostName }}
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: {{ template "fullname" . }}
```
Comment (mgoodness) [inline stable/jenkins/templates/jenkins-master-ingress.yaml]: The nginx controller requires at least a `kubernetes.io/ingress.class: nginx` annotation on the Ingress resource, but supports others as well. Perhaps provide an `ingressAnnotations` value ?

values.yaml:
```
...
ingressAnnotations:
  # kubernetes.io/ingress.class: nginx
  # kubernetes.io/tls-acme: "true"
...
```

jenkins-master-ingress.yaml:
```
...
metadata:
{{- if .Values.ingressAnnotations }}
  annotations:
{{ toYaml .Values.ingressAnnotations | indent 4 }}
{{- end }}
...
```

**Review by mgoodness — CHANGES_REQUESTED** [review]
nginx support requires an annotation on the Ingress. Recommend adding an `ingressAnnotations` value for this purpose.

Would also be nice if NOTES.txt output the Ingress endpoint (`http://{{ .Values.Master.HostName }}`) instead of the generic `kubectl port-forward ...`.

**Review by mgoodness — APPROVED** [review]

**Comment by lachie83** [comment]
@mgoodness We waiting on anything else here?

**Comment by mgoodness** [comment]
@lachie83 if it lgtu it lgtm

**Pull request merged**